### PR TITLE
Chemistry updates

### PR DESCRIPTION
**Pull request by munan**
## Prerequisite checklist

- [x ] My code follows the Athena++ [Style Guide](https://github.com/PrincetonUniversity/athena/wiki/Style-Guide)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation in the [Wiki](https://github.com/PrincetonUniversity/athena/wiki) accordingly.
- [x ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

Please review the [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md) file for detailed guidelines.

## Description
Major changes:
- Make six-ray compatible with periodic boundary condition, modified regression test for that

Minor changes:
- Fixed chemical rates
- Add dependency of INT_CHM after INT_HYD

## Testing and validation
Re-run all chemistry regression tests and passed.

### CODE REVIEW

File: src/task_list/time_integrator.cpp [hunk 1]
```diff
@@ -1026,7 +1026,7 @@ TimeIntegratorTaskList::TimeIntegratorTaskList(ParameterInput *pin, Mesh *pm) {
       } else {
         AddTask(INT_SCLR,CALC_SCLRFLX);
       }
-      AddTask(INT_CHM,INT_SCLR);
+      AddTask(INT_CHM,(INT_HYD|INT_SCLR));
```
Comment (felker) [inline src/task_list/time_integrator.cpp:1029]: I am having trouble following the graph of dependencies here. Was it a bug beforehand?

Comment (munan) [reply src/task_list/time_integrator.cpp:1029]: This did not cause a bug so far. But when I was trying to figure out another bug and read through this, I realised that  the order of INT_HYD and INT_CHM may not be guaranteed if I do not add dependencies here. Since INT_CHM also update the (internal) energy, I think it is better to add the dependency. Let me know if it makes sense.
